docs/process/guidance/feature_flags/index.rst

```diff
@@ -0,0 +1,507 @@
+the feature in `snake_case`.
+
+The feature flag must be of type `bool_flag`.
+
+Feature flags of this category reside in
+`eclipse-score/score:flags`_.
+
+Selection of the implementation for an enabled feature
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A feature may be provided by multiple modules. The selection of the module which is used for a feature happens through
+a feature flag.
+
+The name of the flag is defined as `<feature_name>_implementation`, where `<feature_name>` corresponds to the name of
+the feature in `snake_case`.
+
+The feature flag must be of type `string_list_flag` and shall only allow single values.
+
+Feature flags of this category reside in
+`eclipse-score/score:flags`_.
+
+Configuration of a feature
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To modify behavior of a feature, the documentation in the feature tree may point out several feature flags. Such flags
+are of this category.
+
+The name of the flag is defined as `<feature_name>_<configuration>` where:
+
+- `<feature_name>` corresponds to the name of the feature in `snake_case`
+- `<configuration>` corresponds to the name of the configuration in `snake_case`
+
+Feature flags of this category reside in
+`eclipse-score/score:flags`_.
+
+Implementation specific configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Modules may have configuration options that are implementation specific. Such options are not mentioned in the feature
+tree but inside the documentation of the module itself.
+
+The name of the flag is defined as `<feature_name>_<module_name>_<configuration>` where:
+
+- `<feature_name>` corresponds to the name of the feature in `snake_case`
+- `<module_name>` corresponds to the name of the module in `snake_case`
+- `<configuration>` corresponds to the name of the configuration in `snake_case`
+
+Feature flags of this category reside in a top-level directory called `flags` of the module.
+
+Propagation of feature flags
+----------------------------
+
+To take effect beyond Bazel, the values of features needs to be propagated to
+different programming languages.
+
+Definition of Public APIs
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bazel
+    :caption: some_module/BUILD
+
+    alias(
+        name = "public_lib",
+        actual = select({
+            ":config_some_feature": [":lib"],
+            "//conditions:default": [],
+        }),
+    )
+
+    cc_library(
+        name = "lib",
+        # ...
+    )
+
+Please refrain from trying to use the approach below.
+It will not work, since `select()` cannot be combined with `visibility`.
+
+.. code-block:: bazel
+    :caption: some_module/BUILD
+
+    cc_library(
+        name = "public_lib",
+        visibility = select({
+          ":config_some_feature": ["//visibility:__public__"],
+          "//conditions:default": [],
+        }),
+    )
+
+C++
+^^^
+
+The selection which tests shall be executed depending on a feature values shall
+be done via preprocessor macros in the test.cpp
```

Comment:
suggestion based on 10 years of professional C++ experience: ban the preprocessor and use different cpp files.

Reply:
Using different cpp files means we do link seaming which is equally bad. I'm fine going with config files and constexpr values.